cucoslib/workers/recommender.py

```diff
@@ -135,30 +135,31 @@ def get_topmost_ref_stack(self, list_packages):
             }
         }
         json_response = self.execute_gremlin_dsl(payload)
-        if json_response is not None:
-            ref_stack_full_components = self.get_response_data(json_response, data_default=[])
-        ref_stk = {}
-
-        # Calculate similarity score of all reference stacks vs. input stack
-        for key, val in ref_stack_matching_components[0].items():
-            if key in ref_stack_full_components[0]:
-                denominator = float(max(ref_stack_full_components[0].get(key), len(list_packages)))
-                ref_stk[key] = float(val) / denominator
-        # Get the name of reference stack with topmost similarity score
-        sname = max(ref_stk.keys(), key=(lambda key: ref_stk[key]))
-
-        # Get data of reference stack based on sname above
-        str_gremlin = "g.V().has('vertex_label','Stack').has('sname', sname).valueMap(true);"
-
-        payload = {
-            'gremlin': str_gremlin,
-            'bindings': {
-                'sname': sname
+        if len(json_response.get("result",[{}]).get("data",[{}])[0].items()) > 0:
+            if json_response is not None:
+                ref_stack_full_components = self.get_response_data(json_response, data_default=[])
+            ref_stk = {}
+
+            # Calculate similarity score of all reference stacks vs. input stack
+            for key, val in ref_stack_matching_components[0].items():
+                if key in ref_stack_full_components[0]:
+                    ref_stk[key] = float(val) / float(ref_stack_full_components[0].get(key))
+
+            # Get the name of reference stack with topmost similarity score
+            sname = max(ref_stk.keys(), key=(lambda key: ref_stk[key]))
+
+            # Get data of reference stack based on sname above
+            str_gremlin = "g.V().has('vertex_label','Stack').has('sname', sname).valueMap(true);"
+
+            payload = {
+                'gremlin': str_gremlin,
+                'bindings': {
+                    'sname': sname
+                }
             }
-        }
-        json_response = self.execute_gremlin_dsl(payload)
-        if json_response is not None:
-            return self.get_response_data(json_response, data_default=[])
+            json_response = self.execute_gremlin_dsl(payload)
```

Comment:
@tuxdna that's right. [This condition](https://github.com/fabric8-analytics/fabric8-analytics-worker/pull/10/files#diff-e27424ca7473f0836da5680fa3e9d08dR138) has been added to handle an error condition when there is no reference stack in the system.